src/Storageless/Http/Configuration.php

```diff
@@ -28,33 +28,50 @@
 use PSR7Sessions\Storageless\Http\ClientFingerprint\Configuration as FingerprintConfig;
 
 /** @psalm-immutable */
-final class Configuration
+final readonly class Configuration
 {
+    public const int DEFAULT_IDLE_TIMEOUT   = 43200;
+    public const int DEFAULT_REFRESH_TIME   = 60;
+    public const string DEFAULT_COOKIE_NAME = '__Secure-slsession';
+
     private JwtConfig $jwtConfiguration;
-    private Clock $clock;
     private SetCookie $cookie;
-    /** @var positive-int */
-    private int $idleTimeout = 43200;
-    /** @var positive-int */
-    private int $refreshTime = 60;
-    /** @var literal-string */
-    private string $sessionAttribute = SessionMiddleware::SESSION_ATTRIBUTE;
     private FingerprintConfig $clientFingerprintConfiguration;
 
-    public function __construct(
+    /**
+     * @param positive-int   $idleTimeout
+     * @param positive-int   $refreshTime
+     * @param literal-string $sessionAttribute
+     */
+    private function __construct(
         JwtConfig $jwtConfiguration,
+        private Clock $clock,
+        SetCookie $cookie,
+        private int $idleTimeout,
+        private int $refreshTime,
+        private string $sessionAttribute,
+        FingerprintConfig $clientFingerprintConfiguration,
     ) {
-        $this->jwtConfiguration = clone $jwtConfiguration;
-
-        $this->clock = SystemClock::fromSystemTimezone();
-
-        $this->cookie = SetCookie::create('__Secure-slsession')
-            ->withSecure(true)
-            ->withHttpOnly(true)
-            ->withSameSite(SameSite::lax())
-            ->withPath('/');
+        $this->jwtConfiguration               = clone $jwtConfiguration;
+        $this->cookie                         = clone $cookie;
+        $this->clientFingerprintConfiguration = clone $clientFingerprintConfiguration;
```

Comment:
@Ocramius according to https://github.com/psr7-sessions/storageless/pull/579#discussion_r1274497194 here I don't clone the `Clock` object as it is a service and not a config object.